13/12/2019 14:35:36 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.UsuarioNaoEncontradoException   Metodo: recuperaUmUsuarioEPerfis   Mensagem: Usuario não encontrado   Stack Trace: excecao.UsuarioNaoEncontradoException: Usuario não encontrado
	at servico.impl.UsuarioAppServiceImpl.recuperaUmUsuarioEPerfis_aroundBody0(UsuarioAppServiceImpl.java:21)
	at servico.impl.UsuarioAppServiceImpl$AjcClosure1.run(UsuarioAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.UsuarioAppServiceImpl.recuperaUmUsuarioEPerfis(UsuarioAppServiceImpl.java:17)
	at PrincipalCarro.recuperaUmUsuarioEPerfis_aroundBody0(PrincipalCarro.java:38)
	at PrincipalCarro.recuperaUmUsuarioEPerfis_aroundBody1$advice(PrincipalCarro.java:44)
	at PrincipalCarro.main(PrincipalCarro.java:38)
	at Principal.main(Principal.java:36)

13/12/2019 21:00:24 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.UsuarioNaoEncontradoException   Metodo: recuperaUmUsuarioEPerfis   Mensagem: Usuario não encontrado   Stack Trace: excecao.UsuarioNaoEncontradoException: Usuario não encontrado
	at servico.impl.UsuarioAppServiceImpl.recuperaUmUsuarioEPerfis_aroundBody0(UsuarioAppServiceImpl.java:21)
	at servico.impl.UsuarioAppServiceImpl$AjcClosure1.run(UsuarioAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.UsuarioAppServiceImpl.recuperaUmUsuarioEPerfis(UsuarioAppServiceImpl.java:17)
	at PrincipalViagem.recuperaUmUsuarioEPerfis_aroundBody0(PrincipalViagem.java:40)
	at PrincipalViagem.recuperaUmUsuarioEPerfis_aroundBody1$advice(PrincipalViagem.java:44)
	at PrincipalViagem.main(PrincipalViagem.java:40)
	at Principal.main(Principal.java:40)

14/12/2019 08:27:51 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: org.springframework.dao.EmptyResultDataAccessException   Metodo: recuperaUmClienteEMotoristas   Mensagem: No entity found for query; nested exception is javax.persistence.NoResultException: No entity found for query <==> No entity found for query   Stack Trace: org.springframework.dao.EmptyResultDataAccessException: No entity found for query; nested exception is javax.persistence.NoResultException: No entity found for query
	at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:392)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:216)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:417)
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:59)
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:213)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:147)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy28.recuperaUmClienteEMotoristas(Unknown Source)
	at servico.impl.ClienteAppServiceImpl.recuperaUmClienteEMotoristas_aroundBody8(ClienteAppServiceImpl.java:65)
	at servico.impl.ClienteAppServiceImpl$AjcClosure9.run(ClienteAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.ClienteAppServiceImpl.recuperaUmClienteEMotoristas(ClienteAppServiceImpl.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy29.recuperaUmClienteEMotoristas(Unknown Source)
	at PrincipalCliente.recuperaUmClienteEMotoristas_aroundBody14(PrincipalCliente.java:179)
	at PrincipalCliente.recuperaUmClienteEMotoristas_aroundBody15$advice(PrincipalCliente.java:44)
	at PrincipalCliente.main(PrincipalCliente.java:179)
	at Principal.main(Principal.java:28)
Caused by: javax.persistence.NoResultException: No entity found for query
	at org.hibernate.ejb.QueryImpl.getSingleResult(QueryImpl.java:290)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.orm.jpa.SharedEntityManagerCreator$DeferredQueryInvocationHandler.invoke(SharedEntityManagerCreator.java:360)
	at com.sun.proxy.$Proxy38.getSingleResult(Unknown Source)
	at dao.impl.ClienteDAOImpl.recuperaUmClienteEMotoristas(ClienteDAOImpl.java:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:136)
	... 19 more

14/12/2019 08:29:48 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: recuperaUmMotorista   Mensagem: Motorista não encontrado   Stack Trace: excecao.MotoristaNaoEncontradoException: Motorista não encontrado
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotorista_aroundBody6(MotoristaAppServiceImpl.java:77)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure7.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotorista(MotoristaAppServiceImpl.java:71)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy31.recuperaUmMotorista(Unknown Source)
	at PrincipalMotorista.recuperaUmMotorista_aroundBody12(PrincipalMotorista.java:169)
	at PrincipalMotorista.recuperaUmMotorista_aroundBody13$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:169)
	at Principal.main(Principal.java:32)

14/12/2019 08:30:01 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: recuperaUmMotoristaECarros   Mensagem: Motorista não encontrado   Stack Trace: excecao.MotoristaNaoEncontradoException: Motorista não encontrado
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotoristaECarros_aroundBody8(MotoristaAppServiceImpl.java:85)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure9.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotoristaECarros(MotoristaAppServiceImpl.java:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy31.recuperaUmMotoristaECarros(Unknown Source)
	at PrincipalMotorista.recuperaUmMotoristaECarros_aroundBody16(PrincipalMotorista.java:198)
	at PrincipalMotorista.recuperaUmMotoristaECarros_aroundBody17$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:198)
	at Principal.main(Principal.java:32)

14/12/2019 18:57:51 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: recuperaUmMotorista   Mensagem: Motorista não encontrado   Stack Trace: excecao.MotoristaNaoEncontradoException: Motorista não encontrado
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotorista_aroundBody6(MotoristaAppServiceImpl.java:77)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure7.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotorista(MotoristaAppServiceImpl.java:71)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy31.recuperaUmMotorista(Unknown Source)
	at PrincipalMotorista.recuperaUmMotorista_aroundBody12(PrincipalMotorista.java:169)
	at PrincipalMotorista.recuperaUmMotorista_aroundBody13$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:169)
	at Principal.main(Principal.java:32)

14/12/2019 18:59:10 org.hibernate.engine.jdbc.spi.SqlExceptionHelper logExceptions ERROR: Cannot delete or update a parent row: a foreign key constraint fails (`banco`.`carro`, CONSTRAINT `MOTORISTA_CARRO_FK` FOREIGN KEY (`motorista_id`) REFERENCES `motorista` (`id`) ON DELETE NO ACTION)
14/12/2019 19:24:18 org.hibernate.engine.jdbc.spi.SqlExceptionHelper logExceptions ERROR: Cannot delete or update a parent row: a foreign key constraint fails (`banco`.`carro`, CONSTRAINT `MOTORISTA_CARRO_FK` FOREIGN KEY (`motorista_id`) REFERENCES `motorista` (`id`) ON DELETE NO ACTION)
14/12/2019 19:32:25 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: exclui   Mensagem: Este produto possui lances e não pode ser removido   Stack Trace: excecao.MotoristaNaoEncontradoException: Este produto possui lances e não pode ser removido
	at servico.impl.MotoristaAppServiceImpl.exclui_aroundBody4(MotoristaAppServiceImpl.java:66)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure5.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.exclui(MotoristaAppServiceImpl.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.exclui(Unknown Source)
	at PrincipalMotorista.exclui_aroundBody14(PrincipalMotorista.java:186)
	at PrincipalMotorista.exclui_aroundBody15$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:186)
	at Principal.main(Principal.java:32)

14/12/2019 19:36:25 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.CarroNaoEncontradoException   Metodo: exclui   Mensagem: Este carro possui motoristas e não pode ser removido!   Stack Trace: excecao.CarroNaoEncontradoException: Este carro possui motoristas e não pode ser removido!
	at servico.impl.CarroAppServiceImpl.exclui_aroundBody4(CarroAppServiceImpl.java:65)
	at servico.impl.CarroAppServiceImpl$AjcClosure5.run(CarroAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.CarroAppServiceImpl.exclui(CarroAppServiceImpl.java:60)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy33.exclui(Unknown Source)
	at PrincipalCarro.exclui_aroundBody16(PrincipalCarro.java:199)
	at PrincipalCarro.exclui_aroundBody17$advice(PrincipalCarro.java:44)
	at PrincipalCarro.main(PrincipalCarro.java:199)
	at Principal.main(Principal.java:36)

14/12/2019 19:50:21 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: exclui   Mensagem: Este motorista possui clientes e não pode ser removido!   Stack Trace: excecao.MotoristaNaoEncontradoException: Este motorista possui clientes e não pode ser removido!
	at servico.impl.MotoristaAppServiceImpl.exclui_aroundBody4(MotoristaAppServiceImpl.java:66)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure5.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.exclui(MotoristaAppServiceImpl.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.exclui(Unknown Source)
	at PrincipalMotorista.exclui_aroundBody14(PrincipalMotorista.java:186)
	at PrincipalMotorista.exclui_aroundBody15$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:186)
	at Principal.main(Principal.java:32)

14/12/2019 20:11:51 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.CarroNaoEncontradoException   Metodo: exclui   Mensagem: Este carro possui viagens e não pode ser removido!   Stack Trace: excecao.CarroNaoEncontradoException: Este carro possui viagens e não pode ser removido!
	at servico.impl.CarroAppServiceImpl.exclui_aroundBody4(CarroAppServiceImpl.java:65)
	at servico.impl.CarroAppServiceImpl$AjcClosure5.run(CarroAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.CarroAppServiceImpl.exclui(CarroAppServiceImpl.java:60)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy33.exclui(Unknown Source)
	at PrincipalCarro.exclui_aroundBody16(PrincipalCarro.java:199)
	at PrincipalCarro.exclui_aroundBody17$advice(PrincipalCarro.java:44)
	at PrincipalCarro.main(PrincipalCarro.java:199)
	at Principal.main(Principal.java:36)

14/12/2019 20:13:57 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: exclui   Mensagem: Este motorista possui carros e não pode ser removido!   Stack Trace: excecao.MotoristaNaoEncontradoException: Este motorista possui carros e não pode ser removido!
	at servico.impl.MotoristaAppServiceImpl.exclui_aroundBody4(MotoristaAppServiceImpl.java:66)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure5.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.exclui(MotoristaAppServiceImpl.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.exclui(Unknown Source)
	at PrincipalMotorista.exclui_aroundBody14(PrincipalMotorista.java:186)
	at PrincipalMotorista.exclui_aroundBody15$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:186)
	at Principal.main(Principal.java:32)

14/12/2019 20:23:33 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.CarroJaTemViagensException   Metodo: exclui   Mensagem: Este carro possui viagens e não pode ser removido!   Stack Trace: excecao.CarroJaTemViagensException: Este carro possui viagens e não pode ser removido!
	at servico.impl.CarroAppServiceImpl.exclui_aroundBody4(CarroAppServiceImpl.java:66)
	at servico.impl.CarroAppServiceImpl$AjcClosure5.run(CarroAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.CarroAppServiceImpl.exclui(CarroAppServiceImpl.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy33.exclui(Unknown Source)
	at PrincipalCarro.exclui_aroundBody16(PrincipalCarro.java:199)
	at PrincipalCarro.exclui_aroundBody17$advice(PrincipalCarro.java:44)
	at PrincipalCarro.main(PrincipalCarro.java:199)
	at Principal.main(Principal.java:36)

14/12/2019 20:26:37 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaJaTemCarrosException   Metodo: exclui   Mensagem: Este motorista possui carros e não pode ser removido!   Stack Trace: excecao.MotoristaJaTemCarrosException: Este motorista possui carros e não pode ser removido!
	at servico.impl.MotoristaAppServiceImpl.exclui_aroundBody4(MotoristaAppServiceImpl.java:67)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure5.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.exclui(MotoristaAppServiceImpl.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.exclui(Unknown Source)
	at PrincipalMotorista.exclui_aroundBody14(PrincipalMotorista.java:186)
	at PrincipalMotorista.exclui_aroundBody15$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:186)
	at Principal.main(Principal.java:32)

14/12/2019 20:49:03 org.hibernate.engine.jdbc.spi.SqlExceptionHelper logExceptions ERROR: Data truncation: Data too long for column 'email' at row 1
14/12/2019 20:49:03 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: org.springframework.dao.DataIntegrityViolationException   Metodo: inclui   Mensagem: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.DataException: could not execute statement <==> could not execute statement <==> Data truncation: Data too long for column 'email' at row 1   Stack Trace: org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.DataException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:252)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:214)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:417)
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:59)
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:213)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:147)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy30.inclui(Unknown Source)
	at servico.impl.MotoristaAppServiceImpl.inclui_aroundBody0(MotoristaAppServiceImpl.java:40)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure1.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.inclui(MotoristaAppServiceImpl.java:27)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.inclui(Unknown Source)
	at PrincipalMotorista.inclui_aroundBody4(PrincipalMotorista.java:102)
	at PrincipalMotorista.inclui_aroundBody5$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:102)
	at Principal.main(Principal.java:32)
Caused by: org.hibernate.exception.DataException: could not execute statement
	at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:71)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:49)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:125)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:110)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:136)
	at org.hibernate.id.IdentityGenerator$GetGeneratedKeysDelegate.executeAndExtract(IdentityGenerator.java:96)
	at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:58)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2975)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3487)
	at org.hibernate.action.internal.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:81)
	at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:377)
	at org.hibernate.engine.spi.ActionQueue.addResolvedEntityInsertAction(ActionQueue.java:214)
	at org.hibernate.engine.spi.ActionQueue.addInsertAction(ActionQueue.java:194)
	at org.hibernate.engine.spi.ActionQueue.addAction(ActionQueue.java:178)
	at org.hibernate.event.internal.AbstractSaveEventListener.addInsertAction(AbstractSaveEventListener.java:321)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:286)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:192)
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:125)
	at org.hibernate.ejb.event.EJB3PersistEventListener.saveWithGeneratedId(EJB3PersistEventListener.java:78)
	at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:208)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:151)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:78)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:853)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:827)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:831)
	at org.hibernate.ejb.AbstractEntityManagerImpl.persist(AbstractEntityManagerImpl.java:875)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:289)
	at com.sun.proxy.$Proxy27.persist(Unknown Source)
	at dao.impl.MotoristaDAOImpl.inclui(MotoristaDAOImpl.java:23)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:136)
	... 25 more
Caused by: com.mysql.cj.jdbc.exceptions.MysqlDataTruncation: Data truncation: Data too long for column 'email' at row 1
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:104)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:960)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1116)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1066)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1396)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1051)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:133)
	... 61 more

15/12/2019 20:38:06 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.MotoristaNaoEncontradoException   Metodo: recuperaUmMotoristaECarros   Mensagem: Motorista não encontrado   Stack Trace: excecao.MotoristaNaoEncontradoException: Motorista não encontrado
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotoristaECarros_aroundBody8(MotoristaAppServiceImpl.java:91)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure9.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.recuperaUmMotoristaECarros(MotoristaAppServiceImpl.java:87)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy31.recuperaUmMotoristaECarros(Unknown Source)
	at PrincipalMotorista.recuperaUmMotoristaECarros_aroundBody16(PrincipalMotorista.java:206)
	at PrincipalMotorista.recuperaUmMotoristaECarros_aroundBody17$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:206)
	at Principal.main(Principal.java:32)

15/12/2019 20:39:12 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: excecao.ClienteNaoEncontradoException   Metodo: recuperaUmCliente   Mensagem: Cliente não encontrado   Stack Trace: excecao.ClienteNaoEncontradoException: Cliente não encontrado
	at servico.impl.ClienteAppServiceImpl.recuperaUmCliente_aroundBody6(ClienteAppServiceImpl.java:66)
	at servico.impl.ClienteAppServiceImpl$AjcClosure7.run(ClienteAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.ClienteAppServiceImpl.recuperaUmCliente(ClienteAppServiceImpl.java:60)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:201)
	at com.sun.proxy.$Proxy29.recuperaUmCliente(Unknown Source)
	at PrincipalMotorista.recuperaUmCliente_aroundBody2(PrincipalMotorista.java:88)
	at PrincipalMotorista.recuperaUmCliente_aroundBody3$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:88)
	at Principal.main(Principal.java:32)

15/12/2019 20:40:49 org.hibernate.engine.jdbc.spi.SqlExceptionHelper logExceptions ERROR: Data truncation: Data too long for column 'nome' at row 1
15/12/2019 20:40:49 aspects.AspectoParaLogDeErros efetuaLog ERROR:    Classe do erro: org.springframework.dao.DataIntegrityViolationException   Metodo: inclui   Mensagem: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.DataException: could not execute statement <==> could not execute statement <==> Data truncation: Data too long for column 'nome' at row 1   Stack Trace: org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.DataException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:252)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:214)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:417)
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:59)
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:213)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:147)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy30.inclui(Unknown Source)
	at servico.impl.MotoristaAppServiceImpl.inclui_aroundBody0(MotoristaAppServiceImpl.java:40)
	at servico.impl.MotoristaAppServiceImpl$AjcClosure1.run(MotoristaAppServiceImpl.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at aspects.AspectoParaLogDeErros.efetuaLog(AspectoParaLogDeErros.java:25)
	at servico.impl.MotoristaAppServiceImpl.inclui(MotoristaAppServiceImpl.java:27)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:98)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:266)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy31.inclui(Unknown Source)
	at PrincipalMotorista.inclui_aroundBody4(PrincipalMotorista.java:102)
	at PrincipalMotorista.inclui_aroundBody5$advice(PrincipalMotorista.java:44)
	at PrincipalMotorista.main(PrincipalMotorista.java:102)
	at Principal.main(Principal.java:32)
Caused by: org.hibernate.exception.DataException: could not execute statement
	at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:71)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:49)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:125)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:110)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:136)
	at org.hibernate.id.IdentityGenerator$GetGeneratedKeysDelegate.executeAndExtract(IdentityGenerator.java:96)
	at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:58)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2975)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3487)
	at org.hibernate.action.internal.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:81)
	at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:377)
	at org.hibernate.engine.spi.ActionQueue.addResolvedEntityInsertAction(ActionQueue.java:214)
	at org.hibernate.engine.spi.ActionQueue.addInsertAction(ActionQueue.java:194)
	at org.hibernate.engine.spi.ActionQueue.addAction(ActionQueue.java:178)
	at org.hibernate.event.internal.AbstractSaveEventListener.addInsertAction(AbstractSaveEventListener.java:321)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:286)
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:192)
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:125)
	at org.hibernate.ejb.event.EJB3PersistEventListener.saveWithGeneratedId(EJB3PersistEventListener.java:78)
	at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:208)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:151)
	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:78)
	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:853)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:827)
	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:831)
	at org.hibernate.ejb.AbstractEntityManagerImpl.persist(AbstractEntityManagerImpl.java:875)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:289)
	at com.sun.proxy.$Proxy27.persist(Unknown Source)
	at dao.impl.MotoristaDAOImpl.inclui(MotoristaDAOImpl.java:23)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:136)
	... 25 more
Caused by: com.mysql.cj.jdbc.exceptions.MysqlDataTruncation: Data truncation: Data too long for column 'nome' at row 1
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:104)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:960)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1116)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1066)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1396)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1051)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:133)
	... 61 more

